docs/source/platform/federation.mdx

```diff
@@ -0,0 +1,216 @@
+
+For instance, in order to have a canary deployment, you might maintain two production graphs in the [Schema Registry](https://www.apollographql.com/docs/platform/schema-registry), one called `prod` and one called `prod-canary`. Your deployment of a change to some implementing service named "foo" might look something like this:
+
+1. Check changes in "foo" against `prod` and `prod-canary`:  
+   ```
+   apollo service:check --tag=prod --serviceName=foo && apollo service:check --tag=prod-canary --serviceName=foo
+   ```
+1. Deploy changes to "foo" into your production environment (_Note: This will not roll out changes to the gateway yet_)
+1. Roll over the `prod-canary` graph, containing one gateway container, using
+   ```
+   apollo service:push --tag=prod-canary --serviceName=foo
+   ```
+   (_Note: If composition fails due to intermediate changes to the canary graph, new configuration will not be rolled out_)
+1. Wait for health checks to pass against the canary, watch dashboards, etc.
+1. After the canary is stable, roll out the changes to the rest of production using
+  ```
+  apollo service:push --tag=prod --serviceName=foo
+  ```
+
+Because you can [tag metrics with variants](https://www.apollographql.com/docs/platform/schema-registry/#associating-metrics-with-a-variant) as well, you can use [Apollo Graph Manager](https://engine.apollographql.com) to verify a canary's performance before rolling out changes to the rest of the graph. You can also use a similar strategy with variants to support a variety of other advanced deployment workflows, like blue/green deployments.
+
+### Controlling rollout
+
+To begin, ensure all component services are up and running, and the working directory has a `.env` file containing the appropriate `ENGINE_API_KEY` for your federated service. Then, upload each service to the gateway through `Apollo CLI`:
+
+```bash
+apollo service:push       \
+  --serviceURL={{url}}    \
+  --endpoint={{url}}      \
+  --serviceName={{name}}  \
+  --federated
+```
+
+> Each service needs a unique name, this is how you will identify the service for schema updates
+
+The given `endpoint` will be queried for the new schema which, if compatible with the rest of the schema, will be uploaded to `Apollo Graph Manager`. At this point, any gateways serving this graph will begin to roll over to the new schema.
+
+You can later modify hosted service definitions by rerunning the `apollo service:push` command with the service's name and the new URL.
+
+> What's the difference between `serviceURL` and `endpoint` parameters? The `endpoint` parameter controls the endpoint where the schema will be fetched from at composition, whereas `serviceURL` controls what URL the gateway will query at runtime. This is especially helpful if you have disabled introspection queries in your production services.
```

Comment:
I don't understand the last sentence here, but I love that we're explaining the difference between these two similarly named options.